From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tobias Fischer <tobias.fischer@qut.edu.au>
Date: Fri, 30 Oct 2026 18:23:49 +1000
Subject: [PATCH 4/4] Build without the MeshIO submodule when it is not checked
 out

Source archives (e.g. GitHub's release tarballs) do not contain submodules;
add_subdirectory(MeshIO) then failed even with the plugin off.
---
 plugins/core/IO/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/plugins/core/IO/CMakeLists.txt b/plugins/core/IO/CMakeLists.txt
index aba22267..8fc0dc24 100644
--- a/plugins/core/IO/CMakeLists.txt
+++ b/plugins/core/IO/CMakeLists.txt
@@ -1,4 +1,8 @@
-add_subdirectory( MeshIO )
+# MeshIO is a git submodule: source archives without submodules can still be
+# built (without it)
+if ( EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/MeshIO/CMakeLists.txt" )
+	add_subdirectory( MeshIO )
+endif()
 add_subdirectory( qAdditionalIO )
 add_subdirectory( qCoreIO )
 add_subdirectory( qCSVMatrixIO )
-- 
2.53.0

